### ISSUE TITLE
can't drop partition if directory is missing

### ISSUE THREAD

**Issue by nvtienlg** [questdb/questdb#4107]
### To reproduce

1. remove partition directory in server
2. drop partition using webui
3. error throw: "Partition does not exist"

### QuestDB version:

7.3.8b

### OS, in case of Docker specify Docker and the Host OS:

Ubuntu 22.04

### File System, in case of Docker specify Host File System:

ext4

### Full Name:

Tien Nguyen

### Affiliation:

Freelancer

### Have you followed Linux, MacOs kernel configuration steps to increase Maximum open files and Maximum virtual memory areas limit?

- [X] Yes, I have

### Additional context

It is better to have "force" option for dropping partition already removed (either mistake or data loss) to make db can work again. If facing this situation, db stop working with any queries even outside removed partitions

Thank you

**Comment by ideoma** [comment]
I need to check but I believe "Partition does not exist" means that it's already logically dropped, e.g. missing directory is detected and partition is excluded from the table automatically.

**Comment by nvtienlg** [comment]
Yes i mean that. Directory already removed in server. But Questdb still use that partition on many queries and of course it fails. And it suggests use DROP PARTITION, in turn fails too. I can give you webui information to verify that if need

Thank you

**Comment by ideoma** [comment]
You are right, queries should not fail, thanks for clarifying.

**Comment by arsenalzp** [comment]
Hello,
Any chance to take this issue for those who is new in this project?

**Comment by SpacRocket** [comment]
It appears that the issue arises because openPartitionInfo variable and txn file used in TableReader aren't updated if partition is removed from filesystem.

The reason some calls might fail is due to this line of code and the subsequent error handling.
https://github.com/SpacRocket/questdb/blob/65f84ec2ed4f3daaba0c22863307c1ce514c7732/core/src/main/java/io/questdb/cairo/TableReader.java#L866

Is catching the error and updating aforementioned file is a correct approach to solving this issue?
